pinot-common/src/main/java/com/linkedin/pinot/common/segment/crypt/PinotCrypterFactory.java

```diff
@@ -24,25 +23,20 @@
  * This factory instantiates the PinotCrypter, which participates in encrypting and decrypting files.
  */
 public class PinotCrypterFactory {
-  private static final String CRYPTER_CLASS_CONFIG = "class";
-  private static String _crypterClass;
   public static final Logger LOGGER = LoggerFactory.getLogger(PinotCrypterFactory.class);
 
   // Prevent factory from being instantiated
   private PinotCrypterFactory() {
 
   }
-  public static void init(Configuration config) {
-    _crypterClass = config.getString(CRYPTER_CLASS_CONFIG);
-  }
 
-  public static PinotCrypter create() {
-    String crypterClassName = _crypterClass;
+  public static PinotCrypter create(String crypterClassName) {
     try {
       LOGGER.info("Instantiating PinotCrypter class {}", crypterClassName);
```

Comment:
You may want to also consider an init() method into the crypter class, perhaps with a config class. In that case, an init() method may have to be added to the interface. Alternatively, consider implementing an abstract base class that you can add default methods in the future, and inheriting from that one.